File: lab01/builder/src/main/java/com/example/builder/dto/CharacterBuildRequest.java
package com.example.builder.dto;

import com.example.builder.enums.CharacterType;
import com.example.builder.enums.WeaponType;
import com.fasterxml.jackson.annotation.JsonFormat;

import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class CharacterBuildRequest {
    
    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private CharacterType type;

    private String name;
    private String health;
    private String mana;

    @JsonFormat(shape = JsonFormat.Shape.STRING)
    private WeaponType weapon;
}

File: lab01/builder1/Dto/CharacterBuildRequest.cs
using Builder1.Enums;

namespace Builder1.Dto;

public class CharacterBuildRequest
{
    public CharacterType Type { get; set; }
    public string Name { get; set; } = string.Empty;
    public int? Health { get; set; }
    public int? Mana { get; set; }
    public WeaponType? Weapon { get; set; }
}
